pkg/operator/ceph/cluster/watcher.go

```diff
@@ -65,12 +71,46 @@ func (c *clientCluster) onK8sNode(ctx context.Context, object runtime.Object) bo
 	if !ok {
 		return false
 	}
+
+	// Get CephCluster
+	cluster := c.getCephCluster()
```

Comment:
Let's factor out this new code to a method such as `handleNodeFailure()` so that the new code flow is separated from other handling of the nodes. 
```suggestion
	if err := handleNodeFailure(); err != nil {
	  // handle error, but perhaps allow continuing to other node processing below
	}
```